AfrikaBurn/MapViewController.swift

```diff
@@ -0,0 +1,194 @@
+        
+        self.navigationItem.title = "Map"
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        loadElements()
+    }
+    
+    func loadElements(){
+        
+        for element in allElements {
+            guard let location = element.location else {
+                continue
+            }
+            let anno = BurnAnnotation(coordinate: location);
+            anno.element = element
+            anno.title = element.name
+            anno.image = UIImage(named: "map-pin.png")
+            
+            mapView.addAnnotation(anno);
+        }
+    }
+    
+}
+
+class BurnMapView: MKMapView, MKMapViewDelegate {
+    
+    let overlay = BurnMapOverlay()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    func commonInit() {
+        showsScale = true
+        showsCompass = true
+        delegate = self
+        userTrackingMode = .followWithHeading
+        add(overlay, level: .aboveRoads)
+        setVisibleMapRect(BurnMap.boundingMapRect, animated: false)
+    }
+    func mapView(_ mapView: MKMapView, rendererFor overlay: MKOverlay) -> MKOverlayRenderer {
+        switch overlay {
+        case is BurnMapOverlay: return BurnMapOverlayView(overlay: overlay, overlayImage: Resources.mapOverlayImage)
+        default: return MKOverlayRenderer(overlay: overlay)
+        }
+    }
+}
+
+class BurnMapOverlay: NSObject, MKOverlay {
+    let boundingMapRect: MKMapRect = BurnMap.boundingMapRect
+    var coordinate: CLLocationCoordinate2D = CLLocationCoordinate2DMake(19.751633252193361, -32.326217923051772)
+    
+    func canReplaceMapContent() -> Bool {
+        return true
+    }
+}
+
+class BurnMapOverlayView: MKOverlayRenderer {
+    let overlayImage: UIImage
+    
+    init(overlay: MKOverlay, overlayImage: UIImage) {
+        self.overlayImage = overlayImage
+        super.init(overlay: overlay)
+    }
+    
+    override func draw(_ mapRect: MKMapRect, zoomScale: MKZoomScale, in context: CGContext) {
+        guard let imageRef = overlayImage.cgImage else {
+            return
+        }
+        let theMapRect = overlay.boundingMapRect
+        let theRect = rect(for: theMapRect)
+        
+        context.scaleBy(x: 1, y: -1)
+        context.translateBy(x: 0, y: -theRect.height)
+        context.draw(imageRef, in: theRect)
+    }
+}
+
+
+extension MapViewController: MKMapViewDelegate {
+    
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        // If annotation is not of type RestaurantAnnotation (MKUserLocation types for instance), return nil
+        if !(annotation is BurnAnnotation){
+            return nil
+        }
+        
+        var annotationView = self.mapView.dequeueReusableAnnotationView(withIdentifier: "Pin")
```

Comment:
I think we can also make some improvements here to avoid optionals. It would also be nice to define the reuse identifier in a struct like `struct ReuseIdentifiers { static let pinAnnotation = "burnPinAnnotation"}` but this is not the end of the world.

So if we instead had:

```swift
let annotationView: MKPinAnnotationView
if let reusedAnnotationView = mapView.dequeueReusableAnnotationView(withIdentifier: "Pin") {
    annotationView = reusedAnnotationView
    annotationView.annotation = annotation
} else {
    annotationView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: "Pin")
    annotationView.canShowCallout = true
}
```

This approach means we no longer need to deal with the optionality of the annotation view and thus the chance of a force unwrap is gone :)